qiskit/backends/_sympy_qasmsimulator.py

```diff
@@ -0,0 +1,438 @@
+                qubit0 = operation['qubits'][0]
+                qubit1 = operation['qubits'][1]
+                opname = operation['name'].upper()
+                opparas = operation['params']
+                q0q1tuple = tuple([qubit0, qubit1])
+                _sym_op = SympyQasmSimulator.get_sym_op(opname, q0q1tuple, opparas)
+                self._quantum_state = qapply(_sym_op * self._quantum_state)
+            # Check if measure
+            elif operation['name'] == 'measure':  # ignore the measure
+                pass
+            elif operation['name'] == 'reset':  # supported modification of state
+                qubit = operation['qubits'][0]
+                self._add_qasm_reset(qubit)
+            elif operation['name'] == 'barrier':
+                pass
+            else:
+                backend = globals()['__configuration']['name']
+                err_msg = '{0} encountered unrecognized operation "{1}"'
+                raise SimulatorError(err_msg.format(backend, operation['name']))
+
+        outcomes = []
+        matrix_form = represent(self._quantum_state)
+        shapeN = matrix_form.shape[0]
+        list_form = [matrix_form[i, 0] for i in range(shapeN)]
+
+        pdist = [SympyQasmSimulator._conjugate_square(matrix_form[i, 0]) for i in range(shapeN)]
+        norm_pdist = [float(i)/sum(pdist) for i in pdist]
+
+        for i in range(actual_shots):  # pylint: disable=unused-variable
+            _classical_state_observed = np.random.choice(np.arange(0, shapeN), p=norm_pdist)
+            outcomes.append(bin(_classical_state_observed)[2:].zfill(
+                self._number_of_cbits))
+
+        # Return the results
+        counts = dict(Counter(outcomes))
+        data = {'counts': self._format_result(counts, cl_reg_index, cl_reg_nbits)}
+
+        data['quantum_state'] = np.asarray(list_form)  # consistent with other backends
+        data['classical_state'] = self._classical_state
+        return {'data': data, 'status': 'DONE'}
+
+    def _format_result(self, counts, cl_reg_index, cl_reg_nbits):
+        """Format the result bit string.
+        This formats the result bit strings such that spaces are inserted
+        at register divisions.
+
+        Args:
+            counts (dict): dictionary of counts e.g. {'1111': 1000, '0000':5}
+            cl_reg_index (list): starting bit index of classical register
+            cl_reg_nbits (list): total amount of bits in classical register
+        Returns:
+            dict: spaces inserted into dictionary keys at register boundaries.
+        """
+        fcounts = {}
+        for key, value in counts.items():
+            new_key = [key[-cl_reg_nbits[0]:]]
+            for index, nbits in zip(cl_reg_index[1:],
+                                    cl_reg_nbits[1:]):
+                new_key.insert(0, key[-(index+nbits):-index])
+            fcounts[' '.join(new_key)] = value
+        return fcounts
+
+    @staticmethod
+    def regulate(theta):
```

Comment:
yes, but this kind of refactoring requires us to first finalize the organization of all the simulators. Otherwise, we will have to change it back and forth.

Reply:
I would do this after the whole team agree on how to organize the set of simulators (i.e., the common interface of all simulators), which will take quite a while. Maybe we can make an issue about this?